<!DOCTYPE html>
<html>

<head>
    <meta charset="UTF-8">
</head>

<body>
    <!-- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise -->
    <script>
        try {
            // JS can handle error in try-catch
            throw new Error("JS error");
        } catch (e) {
            console.log(e);
        }

        try {
            // Promise's error can't be catch
            // because error is not thrown into current thread
            // it is going into browser async message queue
            Promise.reject("Promise error");
        } catch (e) {
            console.log(e);
        }
    </script>
</body>

</html>